velox/exec/RowNumber.cpp

```diff
@@ -491,4 +477,46 @@ void RowNumber::spillInput(
   }
 }
 
+void RowNumber::addInputInternal(const RowVectorPtr& input, bool fromSpill) {
+  const auto numInput = input->size();
```

Comment:
Shall we call ensureInputFits to prevent memory memory arbitration? Thanks!

Reply:
It would be nice to add it. I will conduct a more proper way to add it in the follow-up.

Reply:
Leave a TODO here? Thanks!